splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/SetOpOperationIT.java

```diff
@@ -123,6 +129,104 @@ public void testExcept() throws Exception {
         Assert.assertEquals("Wrong Min", 2, rs.getInt(3));
     }
 
+    /**
+     *
+     *     Test wrong results cases where one SetOp branch uses a covered index and the other
+     *     branch uses a base table access.
+     *
+     */
+    @Test
+    public void testSPLICE2079() throws Exception {
+        ResultSet rs;
+/*        rs = methodWatcher.executeQuery(
+                format("select count(*) from \n" +
+                        "(select col1, col2 from foo4 --splice-properties index=foo4_ix\n" +
+                        "intersect\n" +
+                        "select col1, col2 from foo4 --splice-properties index=null\n" +
+                        ") dtab\n",useSpark));
+        Assert.assertTrue("One row expected.",rs.next());
+        Assert.assertEquals("SPLICE-2079 test 1 failed.", 3, rs.getInt(1));*/
+
+        rs = methodWatcher.executeQuery(
+                "select count(*) from \n" +
```

Comment:
This test can test both Spark and control path. By specifying another hint useSpark=%s in the query, and pass useSpark as a parameter like the example that you've commented out right above, you can cover both spark and control path. In fact, this ticket has wrong result only for control path, so it is better to explicitly test both paths and make sure both work fine.